src/NzbDrone.Core/Notifications/Slack/SlackService.cs

```diff
@@ -0,0 +1,109 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Text.RegularExpressions;
+using FluentValidation.Results;
+using NLog;
+using NzbDrone.Core.Rest;
+using NzbDrone.Core.Validation;
+using RestSharp;
+
+namespace NzbDrone.Core.Notifications.Slack
+{
+    public interface ISlackService
+    {
+        void OnDownload(Dictionary<string,string> mapping, SlackSettings settings);
```

Comment:
I'm not a fan of passing a dictionary with the values, seems like extra work to build it, which we avoid in FileNameBuilder.